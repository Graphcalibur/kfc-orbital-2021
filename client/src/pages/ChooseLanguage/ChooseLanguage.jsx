import React from "react";
import CodeLinkBtn from "./components/CodeLinkBtn.jsx";

const ChooseLanguage = () => {
  const langs = ["Python", "C++", "Java"];
  const click = () => {
    const requestOptions = {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      credentials: "include",
      body: JSON.stringify({
        username: "abacaba123",
        password: "SpeedIAmSpeed",
      }),
    };

    fetch("http://localhost:9000/api/authuser", requestOptions);
  };

  const click2 = () => {
      fetch("http://localhost:9000/api/user/abacaba123/testauth", {credentials: "include"})
          .then(response => response.json())
          .then(data => console.log(data));
  };
  return (
    <div>
      {langs.map((lang) => (
        <CodeLinkBtn language={lang} />
      ))}
      <button onClick={click}>login as abacaba123</button>
      <button onClick={click2}>check if abacaba123 is logged in</button>
    </div>
  );
};

export default ChooseLanguage;
