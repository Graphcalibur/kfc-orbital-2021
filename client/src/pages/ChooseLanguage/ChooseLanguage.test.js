import React from "react";
import ChooseLanguage from "./ChooseLanguage";
import { create } from "react-test-renderer";

describe("Testing ChooseLanguage Page", () => {
  test("Test 1", () => {
    const tree = create(<ChooseLanguage />);
    expect(tree).toMatchSnapshot();
  });
});
