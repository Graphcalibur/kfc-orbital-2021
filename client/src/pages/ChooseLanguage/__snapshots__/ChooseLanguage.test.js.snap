// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ChooseLanguage Page Test 1 1`] = `
<div
  className="shadow p-4 mt-3 box container-sm"
>
  <b
    className="text"
  >
    Select the language to practice on:
  </b>
  <div
    className="mt-3 justify-content-evenly row"
  >
    <div
      className="d-grid col-md-4"
    >
      <button
        className="btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <h2>
          Python
        </h2>
      </button>
    </div>
    <div
      className="d-grid col-md-4"
    >
      <button
        className="btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <h2>
          C++
        </h2>
      </button>
    </div>
    <div
      className="d-grid col-md-4"
    >
      <button
        className="btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <h2>
          All Languages
        </h2>
      </button>
    </div>
  </div>
</div>
`;
